controller/src/main/java/io/pravega/controller/util/Config.java

```diff
@@ -172,7 +172,7 @@
             "retention.frequency.minutes", 30, "retention.frequencyMinutes");
 
     public static final Property<Integer> PROPERTY_RETENTION_BUCKET_COUNT = Property.named(
-            "retention.bucket.count", 1, "retention.bucketCount");
+            "retention.bucket.count", 5, "retention.bucketCount");
```

Comment:
I have asked for revisiting these defaults, specially the watermarking one that look quite high to me. Still, I don't know if just changing it to another value is what we want. For instance, for Segment Containers there is a "rule of thumb" to align the number of Segment Containers to the number of cores available (e.g., 1/2 containers per core). Wondering if we can have i) a deeper understanding of the implications of changing this, ii) provide a "rule of thumb" for its configuration, and iii) is there any problem on changing this value on a live cluster?

Reply:
the default `PROPERTY_WATERMARKING_THREAD_COUNT` is `10` and `PROPERTY_RETENTION_THREAD_COUNT` is `1`. The idea behind these new number were to keep the `Bucket_count` = `Thread_count*5`. For the problem part, we need to run system tests if the new number are causing any issue or not.

Reply:
I don't see why number of threads in the thread pool need to be aligned with number of buckets.... I think the ask in the issue was to use a more reasonable bucket count instead of corner values like 1 or 100...

Reply:
Also, from my end considering "reasonable" or not a value implies that we have some understanding of what the value should be that can be translated into a "rule of thumb" when we deploy larger or smaller Pravega cluster. Therefore, apart from changing the value in itself, I would like to see what is our understanding about this value and perhaps add a note either in the `config.properties` or any other suitable place of the documentation instructing users what they have to consider to change this value depending on the cluster size they want to deploy.